src/form-sections/date.ts

```diff
@@ -0,0 +1,339 @@
+import {
+  LitElement,
+  html,
+  PropertyValues,
+  nothing,
+  css,
+  TemplateResult,
+} from 'lit';
+import { customElement, property, query } from 'lit/decorators.js';
+import '@internetarchive/donation-form-section';
+
+import type { MonthlyPlan } from '../models/plan';
+import '../presentational/ia-button';
+import type { IauxButton } from '../presentational/ia-button';
+import '../presentational/mgc-update-status';
+
+@customElement('ia-mgc-edit-date')
+export class MGCEditPlanDate extends LitElement {
+  @property({ type: Object }) plan?: MonthlyPlan;
+
+  @property({ type: String }) newDate?: string;
+
+  @property({ type: Boolean, reflect: true }) currentlyEditing: boolean = false;
+
+  @property({ type: Boolean }) allowEditing: boolean = false;
+
+  @property({ type: String }) updateMessage: string = '';
+
+  @property({ type: String }) errorMessage: string = '';
+
+  @property({ type: String }) warningMessage: string = '';
+
+  @property({ type: String }) updateStatus: 'success' | 'fail' | '' = '';
+
+  @query('form') form!: HTMLFormElement;
+
+  @query('form ia-button#edit-date') formSubmitButton!: IauxButton;
+
+  @query('form input[name="edit-date"]')
+  dateInput!: HTMLInputElement;
+
+  willUpdate(changed: PropertyValues) {
+    if (this.dateInput && changed.has('plan') && this.plan) {
+      this.dateInput.setAttribute('min', this.minDate);
+      this.dateInput.setAttribute('max', this.maxDate);
+    }
+  }
+
+  updated(changed: PropertyValues) {
+    if (changed.has('currentlyEditing') && this.currentlyEditing) {
+      this.dateInput.focus();
+    }
+  }
+
+  protected render() {
+    return html`
+      <section>
+        <donation-form-section
+          badgemode="hidebadge"
+          headline="Change next donation date"
+        >
```

Comment:
Q: What is the `donation-form-section` component handling here?

Reply:
the styling of the form block 😎